src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientAdapterFactory.java

```diff
@@ -74,33 +89,41 @@ protected void unbindGraphqlClient(GraphqlClient graphqlClient, Map<?, ?> proper
     @SuppressWarnings("unchecked")
     @Override
     public <AdapterType> AdapterType getAdapter(Object adaptable, Class<AdapterType> type) {
+        LOGGER.debug("Try to get a new graphql client from a resource");
         if (!(adaptable instanceof Resource) || type != GraphqlClient.class) {
             return null;
         }
 
-        Resource res = (Resource) adaptable;
-
-        ConfigurationBuilder configBuilder = res.adaptTo(ConfigurationBuilder.class);
-        ValueMap properties = configBuilder.name(CONFIG_NAME).asValueMap();
-
-        String identifier = properties.get(GraphqlClientConfiguration.CQ_GRAPHQL_CLIENT, String.class);
-        if (identifier == null) {
-            identifier = readFallbackConfiguration(res);
-        }
-
-        if (StringUtils.isEmpty(identifier)) {
-            LOGGER.error("Could not find {} property for given resource at {}", GraphqlClientConfiguration.CQ_GRAPHQL_CLIENT, res
-                .getPath());
+        try (ResourceResolver serviceResolver = resolverFactory.getServiceResourceResolver(authInfo)) {
```

Comment:
Just an idea, but couldn't we write the code so that it uses the service user if it exists and is properly configured, or fallbacks to the default "no service user" implementation? In this way, the client would work as before for on-prem customers, and would use the service user if it's properly configured. I'm a bit worried that this change might be complicated to setup for many users, so if this is possible, and it could be a good compromise.